### PR TITLE
Add Bashas' spider

### LINKED ISSUE #3506: Bashas'
https://www.bashas.com/locations/

### PR DESCRIPTION
**Pull request by cmecklenborg**
Fixes #3506

### CODE REVIEW

**Comment by cmecklenborg** [comment]
Because of non-ascii characters in the store names, the geojson output contains escaped unicode characters under the "name" attribute, e.g. `Bashas\u2019 Supermarket: Power & McDowell`. Is this a concern? If so, what is the preferred way to normalize these characters?

**Comment by iandees** [comment]
> the geojson output contains escaped unicode characters under the "name" attribute

That's ok. JSON parsers will unescape those into unicode codepoints properly.

File: locations/spiders/bashas.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    def parse_store(self, response):
+        script = response.xpath(
+            "//script[@id='wpsl-js-js-extra']/text()"
+        ).extract_first()
+        open_brace = script.find('"locations":[{') + 13
+        close_brace = script.rfind("}]}") + 1
+        store_json = json.loads(script[open_brace:close_brace])
+        hours = response.xpath("//table[@class='wpsl-opening-hours']//text()").extract()
+
+        item = DictParser.parse(store_json)
```
Comment (iandees) [inline locations/spiders/bashas.py:29]: It looks like the DictParser is picking up an address and setting it as `addr_full` when it should be `street_address`. Can you add a `item["street_address"] = item.pop("addr_full", None)` somewhere after this line?

Comment (cmecklenborg) [reply locations/spiders/bashas.py:29]: I made this change. I noticed the same problem was happening with a spider I added yesterday (Gelsons), so I made the change there too.

**Pull request merged**